### DISCUSSION TITLE
Need Help Retrieving ASIN Images and Titles for Negative Targeting via API

### DISCUSSION THREAD

**Discussion by AlexUnitix** [amzn/ads-advanced-tools-docs#280] [General]
Hello,

We have encountered an issue while working with search term reports. We use these reports to manage the negative targeting of low-quality search terms. In the Amazon Ads dashboard, we can see the ASINs directly along with their images and titles, which helps us quickly identify which products need to be added to negative targeting.

However, I haven't found a way to achieve this via the API. Could you please assist us with the following:

How can we retrieve ASINs along with their images and titles through the API for the purpose of negative targeting?
Is there a way to decode the identifiers and brands through the API?
Thank you for your help!

**Comment by elizafarley** [comment]
You can use the [POST /product/metadata](https://advertising.amazon.com/API/docs/en-us/product-metadata#tag/Product-Selector/operation/ProductMetadata) API to pull related metadata for a specified ASIN. 

If you need for info that what is provided there, you could try using the Selling Partner API in combination with the Ads API to pull details on your ASINs. For example, the Selling Partner [catalog API](https://developer-docs.amazon.com/sp-api/docs/catalog-items-api-v2022-04-01-reference) might be helpful to you.

Resolved brand identifiers she be available via the [negative target list APIs](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod) and [export API](https://advertising.amazon.com/API/docs/en-us/exports).

**Reply by AlexUnitix** [reply]
Thank you for the response.

We have tried the POST /product/metadata API, but it only provides information for products that are in the client's inventory. The search term report, however, provides information for a wide range of ASINs, not just those in our inventory.

We need a way to retrieve metadata (including images and titles) for these ASINs from the search term report, which are not necessarily in our inventory.

**Reply by phanhiep2603** [reply]
> You can use the [POST /product/metadata](https://advertising.amazon.com/API/docs/en-us/product-metadata#tag/Product-Selector/operation/ProductMetadata) API to pull related metadata for a specified ASIN.
> 
> If you need for info that what is provided there, you could try using the Selling Partner API in combination with the Ads API to pull details on your ASINs. For example, the Selling Partner [catalog API](https://developer-docs.amazon.com/sp-api/docs/catalog-items-api-v2022-04-01-reference) might be helpful to you.
> 
> Resolved brand identifiers she be available via the [negative target list APIs](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod) and [export API](https://advertising.amazon.com/API/docs/en-us/exports).

Hi,
I am aware of the /product/metadata API to fetch metadata for a single ASIN, but I am unable to retrieve the data for ASINs listed in the variationList.

Could you please help me? Is there any API within the Advertising API that can provide the title or product_name ( title) for all ASINs in variationList under my account?
